{{ cookiecutter.repo_name }}/.pre-commit-config.yaml

```diff
@@ -26,4 +26,30 @@ repos:
     -   id: check-added-large-files
         args: ["--maxkb=5120"]
     -   id: end-of-file-fixer
+        exclude: '\.Rd'
     -   id: trailing-whitespace
+{% if cookiecutter.add_r_precommit_hooks == "Yes" %}
+# R specific hooks: https://github.com/lorenzwalthert/precommit
+-   repo: https://github.com/lorenzwalthert/precommit
+    rev: v0.1.2
+    hooks:
+    -   id: style-files
+        args: [--style_pkg=styler, --style_fun=tidyverse_style]
+    -   id: roxygenize
+    # codemeta must be above use-tidy-description when both are used
+    # -   id: codemeta-description-updated
+    -   id: use-tidy-description
+    -   id: lintr
+        exclude: renv/activate.R
+    -   id: readme-rmd-rendered
+    -   id: parsable-R
+    -   id: no-browser-statement
+-   repo: local
+    hooks:
+    -   id: forbid-to-commit
+        name: Don't commit common R artifacts
+        entry: Cannot commit .Rhistory, .RData, .Rds or .rds.
+        language: fail
+        files: '\.Rhistory|\.RData|\.Rds|\.rds$|\.Renviron'
+        # `exclude: <regex>` to allow committing specific files.
```

Comment:
Do we really want to be so strict to include by default data files `.RData`, `.Rds` and `.rds`? If so, `.rda` is missing from the list.

However, I do not think we want this. Users may want to commit data or trained models without having to modify the `.pre-commit-config.yaml` (and can exclude them from committing these files via `.gitignore` as usual).

Through the python pre-commit, we check for any large files (over 5 MB) being committed (which will cover both python- and r- generated objects). I think that's enough? 

@nacnudus @avisionh

Reply:
[Hooks can be skipped](https://pre-commit.com/#temporarily-disabling-hooks) with `SKIP=flake8 git commit -m "foo" (if using git at the command line), but I agree that this could be handled by `.gitignore`.

Reply:
These are very good points. Added it to https://github.com/ukgovdatascience/govcookiecutter/pull/16/commits/e13ad5297ebdbfca6f7237c4783fd12a2781acf2 now!